packages/components/src/components/FormInput/ControlledFormInput.tsx

```diff
@@ -0,0 +1,70 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Controller } from "react-hook-form";
+import type { ControllerProps, Control } from "react-hook-form";
+import { FormInput } from "./FormInput";
+import type { FormInputProps } from "./FormInput";
+
+export interface ControlledFormInputProps
+  extends Omit<ControllerProps, "control" | "render">,
+    Omit<FormInputProps, "defaultValue" | "name"> {
```

Comment:
Will controller pass through any of the form input props provided? I see you're spreading the remaining props into controller right now.

Reply:
Right. The props need to spread to the Controller, but I should pull out any FormInput props and spread those on the actual FormInput. Good catch.

Reply:
@RohitRajendran after digging into this more, I ended up using the necessary props for Controller, and then spread everything else to the input. That way we catch any extra props that need to be added directly to that. Controller has all that it needs.